src/analysis/special_functions/log/deriv.lean

```diff
@@ -268,4 +268,59 @@ begin
       by simpa [pow_succ'] using mul_le_of_le_one_right (pow_nonneg (abs_nonneg x) i) (le_of_lt h) }
 end
 
+/-- Power series expansion of `log(1 + x) - log(1 - x)` for `|x| < 1`. -/
+lemma has_sum_log_sub_log_of_abs_lt_1 {x : ℝ} (h : |x| < 1) :
+  has_sum (λ k : ℕ, (2 : ℝ) * (1 / (2 * k + 1)) * (x ^ (2 * k + 1))) (log (1 + x) - log(1 - x)) :=
+begin
+  have h₁, from has_sum_pow_div_log_of_abs_lt_1 h,
+  have h₂, from has_sum_pow_div_log_of_abs_lt_1 (eq.trans_lt (abs_neg x) h),
+  replace h₂ := (has_sum_mul_left_iff  (λ h : ( -1 = (0:ℝ)), one_ne_zero $ neg_eq_zero.mp h)).mp h₂,
+  rw [neg_one_mul, neg_neg, sub_neg_eq_add 1 x] at h₂,
+  have h₃, from has_sum.add h₂ h₁,
+  rw [tactic.ring.add_neg_eq_sub] at h₃,
+  let term := (λ n : ℕ, ((-1) * ((-x) ^ (n + 1) / ((n : ℝ) + 1)) + (x ^ (n + 1) / (n + 1)))),
+  let two_mul := (λ (n : ℕ), (2 * n)),
+  rw ←function.injective.has_sum_iff (nat.mul_right_injective two_pos) _ at h₃,
+  { suffices h_term_eq_goal :
+    (term ∘ two_mul) = (λ k : ℕ, 2 * (1 / (2 * (k : ℝ) + 1)) * x ^ (2 * k  + 1)),
+    by {rw h_term_eq_goal at h₃, exact h₃},
+    apply funext,
+    intro n,
+    rw [function.comp_app],
+    dsimp only [two_mul, term],
+    rw odd.neg_pow (⟨n, rfl⟩ : odd (2 * n + 1)) x,
+    rw [neg_one_mul, neg_div, neg_neg, nat.cast_mul, nat.cast_two],
+    ring },
+  { intros m hm,
+    rw [range_two_mul, set.mem_set_of_eq] at hm,
+    rw [even.neg_pow (nat.even_succ.mpr hm), nat.succ_eq_add_one, neg_one_mul, neg_add_self] },
+end
+
+/--
+A lemma expanding the log of a fraction. The goal is to use it in `power_series_log_succ_div`.
+-/
+lemma log_succ_div_eq_log_sub (a : ℝ) (h : 0 < a) :
+  log ((a + 1) / a) = log (1 + 1 / (2 * a + 1)) - log (1 - 1 / (2 * a + 1)) :=
+begin
+  have h₀: (2 : ℝ) * a + 1 ≠ 0, by { linarith, },
+  have h₁: a ≠ 0 := ne_of_gt h,
+  rw ← log_div,
+  suffices g, from congr_arg log g,
+  all_goals { field_simp, },
+  all_goals { linarith, },
+end
+
+/-- Power series expansion of `log ((a + 1) / a)`. -/
+theorem power_series_log_succ_div (a : ℝ) (h : 0 < a) : has_sum (λ (k : ℕ),
+  (2 : ℝ) * (1 / (2 * k + 1)) * ((1 / (2 * a + 1)) ^ (2 * k + 1)))
+  (log ((a + 1) / a)) :=
+ begin
+  have h₁ : |1 / (2 * a + 1)| < 1, by
+  { rw [abs_of_pos, div_lt_one],
+    any_goals {linarith},
+    {refine div_pos one_pos _, linarith, }, },
```

Comment:
```suggestion
    { refine div_pos one_pos _, linarith, }, },
```
I think the nested commas look really ugly, and I prefer to end proofs without a trailing comma, but this isn't against the style guide, so feel free to ignore.